Habuild patch 1

habuild opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

I ended up on this page while searching for 'notify.persistent notification'
I was following this example and found it lacking, and some of the markdown commands either non working or hard to follow.

The previous Sample Text is in the form of json which doesn't work in dev tools and sets the service back to null when entered.
This has been changed to yaml and the related picture has been updated to a more current version.

Newline /n doesn't appear to function which requires double spacing of lines to solve.
see yaml markdown example.

Added example Markdown and HTML variable notification and related image,
This should make it clearer for new users as to see what is allowed and can be achieved.

- [X] Added example and image to the current documentation (`current` branch).


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [X] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

habuild committed:
Update persistent_notification.markdown

Added markdown variable example and related image

habuild committed:
Update persistent_notification.markdown

habuild committed:
Update persistent_notification.markdown

habuild committed:
Update persistent_notification.markdown

habuild committed:
Update persistent_notification.markdown

habuild committed:
Update persistent_notification.markdown

habuild committed:
Update persistent_notification.markdown

frenck commented on source/_integrations/persistent_notification.markdown:
```diff
@@ -100,3 +100,29 @@ Choose the **Services** tab from the **Developer Tools** sidebar item, then sele
+
+This is the output.
+
+![persistant_notification_msgCapture](https://user-images.githubusercontent.com/36288425/191888076-0895e15a-2cc2-4e31-9755-c758135b108e.PNG)
```
Images should be included in the repository.

habuild replied:
noted. moved.

frenck commented on source/_integrations/persistent_notification.markdown:
````diff
@@ -100,3 +100,29 @@ Choose the **Services** tab from the **Developer Tools** sidebar item, then sele
 ```
 This will create the notification entry shown above.
+
+### Example Markdown variable notification
````
What is this example adding? As in, I'm not sure what you are trying to show?
Please update the PR description to elaborate / motivate the change.

habuild replied:
updated proposed changes

frenck commented on source/_integrations/persistent_notification.markdown:
```diff
@@ -100,3 +100,29 @@ Choose the **Services** tab from the **Developer Tools** sidebar item, then sele
+data:
+  notification_id: os-agent            ## <<< this can be whatever 12343232
+  title: "There is an update in Os-Agent!"
+  message: >-
```
The example title says this is a Markdown example, but it contains mainly HTML. Please remove the HTML.

habuild replied:
https://daringfireball.net/projects/markdown/syntax#p

That is the link from the page we are discussing.

the current example is. Go to dev tools and enter something like

```json
{
  "notification_id": "1234",
  "title": "Sample notification",
  "message": "This is a sample text"
}
```
I didn't want to remove that.

however
You can't actually do that in dev tools - services with `Persistent Notification: Create`. 
It will revert back to a blank service. 

I also found the markdown not to be working correctly. newline  `\n` doesn't seem to work



I couldn't get a lot of the markdown to work. Whereas the html would. The rest is my ignorance I should have listed it as a html example. I could do both, although not sure html is allowed(or promoted).

Try this in dev tools, I am not great at markdown, this doesn't seem to me correct though.
```
service: persistent_notification.create
data:
  notification_id: frenck
  message: >-
    Agent "{{ states('sensor.os_agent_version_github') }}" is
    >available

    ----------

    1.  Bird
    2.  McHale
    3.  Parish

    1.  Bird

    2.  McHale

    3.  Parish


    * Bird

    *  Magic
    *  Bird

    * Magic

    ---------

    <ul>
    <li>Bird</li>
    <li>Magic</li>
    </ul>

    ***    

    * * *

    - - -

    ------------------------------------


    [Latest](https://github.com/home-assistant/os-agent/releases/latest)\n


    **wget https://github.com/home-assistant/os-agent/releases/download/** "{{
    states('sensor.os_agent_version_github') }}"/os-agent_"{{
    states('sensor.os_agent_version_github') }}"_linux_aarch64.deb\n
    sudo dpkg \n -i os-agent_"{{ states('sensor.os_agent_version_github')
    }}"_linux_aarch64.deb
  title:  Hi Frenck tyvm


```

frenck commented on source/_integrations/persistent_notification.markdown:
````diff
@@ -100,3 +100,29 @@ Choose the **Services** tab from the **Developer Tools** sidebar item, then sele
+```yaml
+service: persistent_notification.create
+data:
+  notification_id: os-agent            ## <<< this can be whatever 12343232
````
```suggestion
  notification_id: os-agent
```

frenck commented on source/_integrations/persistent_notification.markdown:
```diff
@@ -100,3 +100,29 @@ Choose the **Services** tab from the **Developer Tools** sidebar item, then sele
+service: persistent_notification.create
+data:
+  notification_id: os-agent            ## <<< this can be whatever 12343232
+  title: "There is an update in Os-Agent!"
```
Why is the OS Agent used in this example? Is there a specific reason?

habuild replied:
No specific reason other than, that is what I was making for myself at the time, when I was following the docs.

This might help with the context
https://community.home-assistant.io/t/persisent-notifications-and-critical-notifications/464484

habuild committed:
Update source/_integrations/persistent_notification.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

habuild committed:
Add files via upload

Image related to proposed example changes in  /_integrations/persistent_notification.markdown

habuild committed:
Add files via upload

update sample text image to reflect the current view in frontend when received.

habuild committed:
Delete persistant_notification_msg_capture.PNG

changing for better example image

habuild committed:
Add files via upload

image replacement of persistant_notification_msg_capture.PNG

habuild committed:
Update persistent_notification.markdown

Change non-working json "Sample" example text to working yaml example. 
Updated "Sample text image" to current received message.

Added
Markdown and HTML examples to get around non working commands. eg.  `/n newline`
More extensive Markdown and HTML yaml examples.
Added corresponding image.

habuild committed:
Update persistent_notification.markdown

habuild committed:
Update persistent_notification.markdown

typo choosen corrected
